maintenance/utils/Status.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+class MaintenanceStatus {
+	private $cacheKey = null;
+
+	private $STARTED = 'STARTED';
+	private $ABORTED = 'ABORTED';
+	private $FINISHED = 'FINISHED';
+	private $TTL = 7200; // 2h in seconds
+
+	public function __construct($taskName) {
+		$this->cacheKey = $taskName . '_status';
```

Comment:
This should probably be sharded by wiki ID. If you wrap this in `wfMemcKey` it will do it for you.

Actually that depends on whether the task is per-wiki or global. So this warrants some further discussion.